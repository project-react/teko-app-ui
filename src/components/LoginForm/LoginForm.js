import React, { useState } from 'react';
import Avatar from '@material-ui/core/Avatar';
import Button from '@material-ui/core/Button';
import CssBaseline from '@material-ui/core/CssBaseline';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import Checkbox from '@material-ui/core/Checkbox';
import Link from '@material-ui/core/Link';
import Grid from '@material-ui/core/Grid';
import Box from '@material-ui/core/Box';
import LockOutlinedIcon from '@material-ui/icons/LockOutlined';
import Typography from '@material-ui/core/Typography';
import Container from '@material-ui/core/Container';
import { MaterialInputForm } from 'components/shared/MaterialInputForm';
import { InputText } from 'components/shared/InputText';
import { validatorHelper } from 'helpers/validator';
import Auth from 'services/auth';
import { Link as RouteLink } from 'react-router-dom';
import swal from 'sweetalert';
import CircularProgress from '@material-ui/core/CircularProgress';

export default function LoginForm(props) {
  const classes = MaterialInputForm.useStyles();
  document.title = 'Login';
  const ToolHandlingForm = () => {
    return (
      <div>
        <FormControlLabel
          control={<Checkbox value="remember" color="primary" />}
          label="Remember me"
        />
        <Button
          type="submit"
          fullWidth
          variant="contained"
          color="primary"
          className={classes.submit}
          onClick={submit}
        >
          Login
        </Button> 
        <Grid container>
          <Grid item xs>
            <Link to="/ResetPassword" variant="body2" component={RouteLink}>
              Forgot password?
            </Link>
          </Grid>
          <Grid item>
            <Link to="/Register" variant="body2" component={RouteLink}>
              {' '}
              Register
            </Link>
          </Grid>
        </Grid> 
      </div>
    )
  }

  const ToolLoadingForm = () => {
    return (
      <Box mt={5}>
          <CircularProgress disableShrink />
      </Box>
    )
  }

  let ToolHandlingFormChange = ''; 

  let ToolLoadingFormChange = ""; 

  const [usernameField, setUsernameField] = useState({
    value: '',
    error: '',
    isError: false,
  });

  const [passwordField, setPasswordField] = useState({
    value: '',
    error: '',
    isError: false,
  });

  const [loadingField, setLoadingField] = useState({
    isLoading : false, 
  })  

  const submit = e => {
    e.preventDefault();
    let validatorsField = ['username', 'password'];
    if (
      validatorHelper.isFormValid(validatorHelper.validators, validatorsField)
    ) {
      setLoadingField({isLoading: true}); 
      const data = {
        username: usernameField.value,
        password: passwordField.value,
      };
      Auth.login(data)
        .then(res => {
          let expiredTime = Date.now() + 1800000;
          swal(
            'Hello, ' + usernameField.value,
            'Auto Logout before 30 minutes',
            'success',
          ).then(() => {
            localStorage.setItem('username', usernameField.value);
            localStorage.setItem('token', res.data.token);
            localStorage.setItem('time', expiredTime);
            props.history.push('/Home');
          });
        })
        .catch(err => {
          console.log(err.response.data.message);
          swal('Sorry!', err.response.data.message, 'error');
          setLoadingField({isLoading: false});
        });
    } else {
      swal('Sorry!', 'wrong input format', 'error');
    }
  };

  if(loadingField.isLoading){
    ToolLoadingFormChange = ToolLoadingForm(); 
  } 
  else {
    ToolHandlingFormChange = ToolHandlingForm(); 
  }

  return (
    <Container component="main" maxWidth="xs">
      <CssBaseline />
      <div className={classes.paper}>
        <Avatar className={classes.avatar}>
          <LockOutlinedIcon />
        </Avatar>
        <Typography component="h1" variant="h5">
          Login
        </Typography>
        <form className={classes.form} noValidate>
          <Grid container spacing={2}>
            <Grid item xs={12}>
              <InputText
                autoComplete="fname"
                name="username"
                variant="outlined"
                required
                fullWidth
                id="username"
                label="User Name"
                autoFocus
                type="text"
                textField={usernameField}
                setChange={setUsernameField}
              />
            </Grid>
            <Grid item xs={12}>
              <InputText
                autoComplete="fname"
                name="password"
                variant="outlined"
                required
                fullWidth
                id="password"
                label="Password"
                autoFocus
                type="password"
                textField={passwordField}
                setChange={setPasswordField}
              />
            </Grid>
          </Grid>
          {ToolHandlingFormChange}
        </form>
        {ToolLoadingFormChange}
      </div>
      <Box mt={5}>
        <MaterialInputForm.MadeWithLove />
      </Box>
    </Container>
  );
}
