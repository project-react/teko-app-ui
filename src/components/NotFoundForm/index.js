export { default as NotFoundForm } from './NotFoundForm'