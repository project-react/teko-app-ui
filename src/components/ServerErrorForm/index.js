export {default as ServerErrorForm} from './ServerErrorForm'