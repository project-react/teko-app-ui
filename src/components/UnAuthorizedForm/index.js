export {default as UnAuthorizedForm} from './UnAuthorizedForm'