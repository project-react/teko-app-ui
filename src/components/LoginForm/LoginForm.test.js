import React from 'react';
import { shallow } from 'enzyme';
import { LoginForm } from 'components/LoginForm';
import CssBaseline from '@material-ui/core/CssBaseline';

describe('Test Login Form', () => {
  it('Create snapshot', () => {
    const component = shallow(<LoginForm />);
    expect(component).toMatchSnapshot();
  });
});

describe('First React component test with Enzyme', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<LoginForm />);
    expect(wrapper.find(<CssBaseline />)).toBeDefined();
  });
});
