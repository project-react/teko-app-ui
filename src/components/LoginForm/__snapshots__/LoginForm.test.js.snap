// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Login Form Create snapshot 1`] = `
<WithStyles(ForwardRef(Container))
  component="main"
  maxWidth="xs"
>
  <CssBaseline />
  <div
    className="makeStyles-paper-1"
  >
    <WithStyles(ForwardRef(Avatar))
      className="makeStyles-avatar-2"
    >
      <LockOutlinedIcon />
    </WithStyles(ForwardRef(Avatar))>
    <WithStyles(ForwardRef(Typography))
      component="h1"
      variant="h5"
    >
      Login
    </WithStyles(ForwardRef(Typography))>
    <form
      className="makeStyles-form-3"
      noValidate={true}
    >
      <WithStyles(ForwardRef(Grid))
        container={true}
        spacing={2}
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={12}
        >
          <InputText
            autoComplete="fname"
            autoFocus={true}
            fullWidth={true}
            id="username"
            label="User Name"
            name="username"
            required={true}
            setChange={[Function]}
            textField={
              Object {
                "error": "",
                "isError": false,
                "value": "",
              }
            }
            type="text"
            variant="outlined"
          />
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={12}
        >
          <InputText
            autoComplete="fname"
            autoFocus={true}
            fullWidth={true}
            id="password"
            label="Password"
            name="password"
            required={true}
            setChange={[Function]}
            textField={
              Object {
                "error": "",
                "isError": false,
                "value": "",
              }
            }
            type="password"
            variant="outlined"
          />
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
      <WithStyles(WithFormControlContext(ForwardRef(FormControlLabel)))
        control={
          <ForwardRef(WithStyles)
            color="primary"
            value="remember"
          />
        }
        label="Remember me"
      />
      <WithStyles(ForwardRef(Button))
        className="makeStyles-submit-4"
        color="primary"
        fullWidth={true}
        onClick={[Function]}
        type="submit"
        variant="contained"
      >
        Login
      </WithStyles(ForwardRef(Button))>
      <WithStyles(ForwardRef(Grid))
        container={true}
      >
        <WithStyles(ForwardRef(Grid))
          item={true}
          xs={true}
        >
          <WithStyles(ForwardRef(Link))
            component={[Function]}
            to="/ResetPassword"
            variant="body2"
          >
            Forgot password?
          </WithStyles(ForwardRef(Link))>
        </WithStyles(ForwardRef(Grid))>
        <WithStyles(ForwardRef(Grid))
          item={true}
        >
          <WithStyles(ForwardRef(Link))
            component={[Function]}
            to="/Register"
            variant="body2"
          >
             
            Register
          </WithStyles(ForwardRef(Link))>
        </WithStyles(ForwardRef(Grid))>
      </WithStyles(ForwardRef(Grid))>
    </form>
  </div>
  <Styled(MuiBox)
    mt={5}
  >
    <MadeWithLove />
  </Styled(MuiBox)>
</WithStyles(ForwardRef(Container))>
`;
